templatefitter/templates.py

```diff
@@ -668,8 +668,18 @@ def yield_param_values(self):
 
     @yield_param_values.setter
     def yield_param_values(self, new_values):
-        for template, value in zip(self._template_dict.values(), new_values):
-            template.yield_param_values = value
+        if new_values[0] in self._template_dict:
```

Comment:
if you want to set the yield a specific template by its id, you can use the `templates[id].yield_param_values = new value` syntax